src/frontend/admin/src/components/templates/enrollments/buttons/EnrollmentActions.tsx

```diff
@@ -0,0 +1,138 @@
+import * as React from "react";
+import { useMemo } from "react";
+import { defineMessages, useIntl } from "react-intl";
+import moment from "moment";
+import CancelIcon from "@mui/icons-material/Cancel";
+import CheckCircleIcon from "@mui/icons-material/CheckCircle";
+import ButtonMenu, {
+  MenuOption,
+} from "@/components/presentational/button/menu/ButtonMenu";
+import {
+  useEnrollment,
+  useEnrollments,
+} from "@/hooks/useEnrollments/useEnrollments";
+import { Enrollment } from "@/services/api/models/Enrollment";
+import { useModal } from "@/components/presentational/modal/useModal";
+import { AlertModal } from "@/components/presentational/modal/AlertModal";
+
+const messages = defineMessages({
+  enrollmentActionsLabel: {
+    id: "components.templates.enrollments.buttons.enrollmentActionsButton.enrollmentActionsLabel",
+    description: "Label for the actions button",
+    defaultMessage: "Actions",
+  },
+  enroll: {
+    id: "components.templates.orders.buttons.enrollmentActionsButton.enroll",
+    description: "Label for the enroll enrollment action",
+    defaultMessage: "Enroll",
+  },
+  unroll: {
+    id: "components.templates.orders.buttons.enrollmentActionsButton.unroll",
+    description: "Label for the unroll enrollment action",
+    defaultMessage: "Unroll",
+  },
+  cantEnrollMessage: {
+    id: "components.templates.orders.buttons.enrollmentActionsButton.cantEnrollMessage",
+    description: "Label for the can't enroll tooltip message",
+    defaultMessage:
+      "It's impossible to enroll because the registration date has passed",
+  },
+  validateUnrollModalTitle: {
+    id: "components.templates.orders.buttons.enrollmentActionsButton.validateUnrollModalTitle",
+    description: "Title for the validate unroll modal",
+    defaultMessage: "Unroll this user",
+  },
+  validateUnrollModalMessage: {
+    id: "components.templates.orders.buttons.enrollmentActionsButton.validateUnrollModalMessage",
+    description:
+      "Text for the validate enroll modal message when the enrollment session is open",
+    defaultMessage:
+      "Are you sure you want to unroll? You can register until registration closes. After this date, it will no longer be possible to go back.",
```

Comment:
```suggestion
      "Are you sure you want to unroll? You can enroll until enrollment closes. After this date, it will no longer be possible to revert this action.",
```